dot/core/service.go

```diff
@@ -329,51 +329,50 @@ func (s *Service) safeBabeKill() error {
 }
 
 func (s *Service) handleBabeSession() {
-	for {
-		// wait for BABE epoch to complete
-		<-s.epochDone
+	// wait for BABE epoch to complete
+	for range s.epochDone {
+	}
```

Comment:
change to `<-s.epochDone` ? I don't think there's much need to `range` here

Reply:
the range is there in order to try to avoid race conditions that trigger a panic. Since this is one of the many options where that could happen, I added a for range in order to mitigate the issue and pinpoint the real issue. If in the future we feel more confident, we can revert it back.

Reply:
could you clarify what race condition would be happening here? receiving from a channel will never panic, I'd prefer if you could change it back

Reply:
This is now completely gone since it was redundant and controlling state with channels is really not recommended. Please re-check.